### PR TITLE
SSM parameters in cloudformation (`AWS::SSM::Parameter::`) are not recognized and resolved

### PR DESCRIPTION
**Pull request by sahilshah6196**
When using ssm parameters in cloudforamtion, and `Ref` node is replaced with the corresponding `physical_resource_id.`

Have mention more here:

Related issue:
https://github.com/spulec/moto/issues/3928

### CODE REVIEW

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @sahilshah6196, thanks for the PR! Change looks good to me, just need to make sure that the tests passes on our CI

**Comment by sahilshah6196** [comment]
> Hi @sahilshah6196, thanks for the PR! Change looks good to me, just need to make sure that the tests passes on our CI

I believe I needed to specify the `region_name` [here](https://github.com/spulec/moto/blob/17231741297f96a571e455805f0132529ac117e2/tests/test_cloudformation/test_stack_parsing.py#L518)

File: moto/cloudformation/parsing.py [hunk 1]
```diff
@@ -41,6 +41,7 @@
 from moto.sns import models as sns_models  # noqa
 from moto.sqs import models as sqs_models  # noqa
 from moto.stepfunctions import models as stepfunctions_models  # noqa
+from moto.ssm import models as ssm_models, ssm_backend  # noqa
```
Comment (bblommers) [inline moto/cloudformation/parsing.py:44]: Just realized that the `ssm_backend` specifically refers to the backend for us-east-1, so it's not going to work for other regions. 
Out the top of my head, you might need to use `ssm_models.ssm_backends[region]`

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/cloudformation/parsing.py (hunk 1 above)
Comment (sahilshah6196) [reply moto/cloudformation/parsing.py:44]: I updated the stack to be in `us-east-1`. Reckon that would work? c90cc01

Comment (bblommers) [reply moto/cloudformation/parsing.py:44]: For our test, yeah, but moto users would also be stuck using us-east-1

Comment (sahilshah6196) [reply moto/cloudformation/parsing.py]: @bblommers Everything looks good now?

**Pull request merged**